File: 1.FDP/4° Competencia/Java/Arreglos/Ejercicio3.java
import java.util.Scanner;

public class Ejercicio3 {
	public static void main (String[] args){
		Scanner scan= new Scanner(System.in);
		System.out.println("Calcular el area de un triangulo.\n");
		System.out.println ("El programa se repite 3 veces");
		double matriz[][] = new double [3][3];
		
		// Procedimiento.----------------------------------------------------------------------------
		for (int i=0; i<3; i++) {
			System.out.println("Ingresar base: ");
			matriz[i][0]= scan.nextDouble();
			if (matriz[i][0] > 0) {
				System.out.println("Ingresar altura: ");
				matriz[i][1]= scan.nextDouble();
				if (matriz[i][1] > 0) {
					matriz[i][2]= (matriz[i][0]*matriz[i][1])/2;
					System.out.println("///////////////////////////.");
				} else {
					System.out.println("Error en la altura");
					System.out.println("//////////////////////////");
				}
			} else {
				System.out.println("Error en la base");
				System.out.println("//////////////////////////");
			}
		}
		
		// Impresion de resultados-----------------------------------------------------------------
		System.out.println("Ejercicios realizados");
		for (int i=0; i<3; i++){
			if( matriz[i][1] > 0 && matriz[i][0] > 0 ){
			System.out.println ("Ejercicios " + (i + 1) );
			System.out.println("Area: Base: " + matriz[i][0] + " x Altura: "
			 + matriz[i][1] + "/2 =" + matriz[i][2]);
			}
		}
		scan.close();

		// Creador y fecha------------------------------------------------------------------------
		System.out.println("\n");
		System.out.println("Creador: Jesus A. Mex Damian 16/11/2022");
	}
}

File: 1.FDP/4° Competencia/C++/Arreglos/Ejercicio3.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main(){
	cout << "Calcular el area de un triangulo.\n";
	double matriz [10][3] = {0};
	int i= 0;
	cout << "Se podra repetir hasta 10 veces el proceso.\n";
	int respuesta;

	// Procedimiento.----------------------------------------------------------------------------
	do{
		if ( i < 10 ) {
			cout << "Ingresar la base:";
			cin >> matriz[i][0];
			if (matriz[i][0] > 0){
				cout << "Ingresar la altura:";
				cin >> matriz[i][1];
				if (matriz[i][1] > 0) {
					matriz[i][2]= (matriz[i][1]*matriz[i][0])/2;
					cout << "/////////////////////////////.\n";
				} else {
					cout << "Error en la altura.\n";
					cout << "///////////////////////.\n";	
				}
			} else {
				cout << "Error en la base.\n";
				cout << "///////////////////////.\n";
			}
			i++;
			cout << "Para finalizar oprima 0, para continuar 1: ";
			cin >> respuesta;
			cout << "\n";
		} else {
			respuesta = 0;
		}
	}while(respuesta !=0);
	
	// Impresion de resultados-----------------------------------------------------------------
	for (int i=0;i<10;i++){
		if( matriz[i][1] > 0 && matriz[i][0] > 0  ){
		cout << "Ejercicios " << i +1 << "\n";
		cout << "Area del triangulo: "<< "Base: " << matriz[i][0] << " / Altura: " 
		<< matriz[i][1] << "= " << matriz[i][2] << "\n";
		}	
	}
	
	// Creador y fecha------------------------------------------------------------------------
	cout <<"\n";
	cout <<"Creador: Jesus A. Mex Damian 16/11/2022";
	return 0;
}